src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/SystemIPGlobalProperties.cs

```diff
@@ -0,0 +1,480 @@
+
+namespace System.Net.NetworkInformation
+{
+    internal class SystemIPGlobalProperties : IPGlobalProperties
+    {
+        internal SystemIPGlobalProperties()
+        {
+        }
+
+        internal Interop.IpHlpApi.FIXED_INFO FixedInfo
+        {
+            get
+            {
+                // TODO: #2485: This will need to be moved to PAL.
+                return HostInformationPal.GetFixedInfo();
+            }
+        }
+
+        /// Specifies the host name for the local computer.
+        public override string HostName
+        {
+            get
+            {
+                return FixedInfo.hostName;
+            }
+        }
+
+        /// Specifies the domain in which the local computer is registered.
+        public override string DomainName
+        {
+            get
+            {
+                return FixedInfo.domainName;
+            }
+        }
+
+        /// 
+        /// The type of node.
+        /// 
+        /// <remarks>
+        /// The exact mechanism by which NetBIOS names are resolved to IP addresses
+        /// depends on the node's configured NetBIOS Node Type. Broadcast - uses broadcast
+        /// NetBIOS Name Queries for name registration and resolution.
+        /// PeerToPeer - uses a NetBIOS name server (NBNS), such as Windows Internet
+        /// Name Service (WINS), to resolve NetBIOS names.
+        /// Mixed - uses Broadcast then PeerToPeer.
+        /// Hybrid - uses PeerToPeer then Broadcast.
+        /// </remarks>
+        public override NetBiosNodeType NodeType
+        {
+            get
+            {
+                return (NetBiosNodeType)FixedInfo.nodeType;
+            }
+        }
+
+        /// Specifies the DHCP scope name.
+        public override string DhcpScopeName
+        {
+            get
+            {
+                return FixedInfo.scopeId;
+            }
+        }
+
+        /// Specifies whether the local computer is acting as an WINS proxy.
+        public override bool IsWinsProxy
+        {
+            get
+            {
+                return (FixedInfo.enableProxy);
+            }
+        }
+
+        public override TcpConnectionInformation[] GetActiveTcpConnections()
+        {
+            List<TcpConnectionInformation> list = new List<TcpConnectionInformation>();
+            List<SystemTcpConnectionInformation> connections = GetAllTcpConnections();
+            foreach (TcpConnectionInformation connection in connections)
+            {
+                if (connection.State != TcpState.Listen)
+                {
+                    list.Add(connection);
+                }
+            }
+
+            return list.ToArray();
+        }
+
+        public override IPEndPoint[] GetActiveTcpListeners()
+        {
+            List<IPEndPoint> list = new List<IPEndPoint>();
+            List<SystemTcpConnectionInformation> connections = GetAllTcpConnections();
+            foreach (TcpConnectionInformation connection in connections)
```

Comment:
Same comment as above: no knowledge about percentage of listen / client sockets.